npm/private/utils.bzl

```diff
@@ -379,16 +405,47 @@ def _convert_v6_packages(packages):
 
     return result
 
+def _convert_pnpm_v9_package_dependency_version(name, version):
+    # Convert peer dependency data to rules_js ~v5 format
+    version = _convert_pnpm_v6_v9_version_peer_dep(version)
+
+    return version
+
 def _convert_pnpm_v9_package_dependency_map(deps):
     result = {}
     for name, version in deps.items():
-        # Otherwise assume a reference to another package@version(...maybe peers...)
-        result[name] = _convert_pnpm_v6_v9_version_peer_dep(version)
+        result[name] = _convert_pnpm_v9_package_dependency_version(name, version)
+    return result
+
+def _convert_pnpm_v9_importer_dependency_map(deps):
+    result = {}
+    for name, attributes in deps.items():
+        specifier = attributes.get("specifier")
+        version = attributes.get("version")
+
+        if specifier.startswith("npm:"):
+            # Keep the npm: specifier for aliased dependencies
+            alias, version = version.rsplit("@", 1)
+            version = "npm:{}@{}".format(alias, version)
+        else:
+            # Transition version[(patch)(peer)(data)] to a rules_js version format
+            version = _convert_pnpm_v6_v9_version_peer_dep(version)
 
+        result[name] = version
     return result
 
-# v9 importers are the same as v6 importers
-_convert_v9_importers = _convert_v6_importers
```

Comment:
v9 has less edge cases so I copy+pasted and then deleted some code basically, mixing the two was getting too confusing